src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -70,20 +75,31 @@ public UUID createResource(
               .map(
                   entry ->
                       new MapSqlParameterSource()
-                          .addValue("tracked_resource_id", trackedResourceId)
+                          .addValue("tracked_resource_id", trackedResourceId.getUUID())
                           .addValue("key", entry.getKey())
                           .addValue("value", entry.getValue()))
               .toArray(size -> new MapSqlParameterSource[labels.size()]);
       try {
         jdbcTemplate.batchUpdate(insertLabelSql, sqlParameterSourceList);
       } catch (DuplicateKeyException e) {
-        throw new DuplicateLabelException(
+        throw new DuplicateDbKeyException(
             "Duplicate label found for tracked_resource_id: "
                 + trackedResourceId
                 + " already exists.",
             e);
       }
     }
-    return trackedResourceId;
+    return trackedResourceId.toString();
+  }
+
+  @VisibleForTesting
+  static String serialize(CloudResourceUid resource) {
+    ObjectMapper objectMapper =
+        new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL);
```

Comment:
this needs a comment that it should not be changed since this is how the database will store the json and do serialization. I don't think postgres thinks that `{ "foo": null }` and `{}` are equivalent, even if we definitely want to store the second to save on space since they're logically equivalent in this case..

Reply:
added some comments. 
Actually I think Postgres works fine if we call to_jsonb(string). They are the same in json binary I guess?

Reply:
If that's true, let's not do NON_NULL and just write a more intrusive JanitorDaoTest that asserts that it's ok, like that we don't get back fields with null values in the raw json.

Reply:
Still not resolved. Do we need NON_NULL for jsonb? Add a test if we don't.